File: rb/features.rb
# Valorant SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ValorantFeatures
  def self.make_feature(name)
    case name
    when "base"
      ValorantBaseFeature.new
    when "ratelimit"
      ValorantRatelimitFeature.new
    when "retry"
      ValorantRetryFeature.new
    when "test"
      ValorantTestFeature.new
    when "timeout"
      ValorantTimeoutFeature.new
    else
      ValorantBaseFeature.new
    end
  end
end

File: py/features.py
# Valorant SDK feature factory

from feature.base_feature import ValorantBaseFeature
from feature.test_feature import ValorantTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ValorantBaseFeature(),
        "test": lambda: ValorantTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
